### DISCUSSION TITLE
Snap package almost working

### DISCUSSION THREAD

**Discussion by soumyaDghosh** [Rafostar/clapper#357] [Q&A]
I have almost created the snap package for clapper, but there are still 2 major problems,

1. The snap is not playing any audio for the videos
2. The snap is unable to play videos from the internet

@Rafostar I hope, a little from you will solve this. Thanks

**Comment by Rafostar** [comment]
In order to support video and audio output and every possible codec, you should have:
```sh
gstreamer1.0-plugins-base gstreamer1.0-plugins-good gstreamer1.0-plugins-bad gstreamer1.0-plugins-ugly gstreamer1.0-libav gstreamer1.0-gl gstreamer1.0-pulseaudio
```

**Reply by Rafostar** [reply]
@soumyaDghosh Above packages for 1st issue. For second please share more info what are you trying to play. A direct link to video on the web like `https://example/video.mp4` or video from some webpage like YouTube, Twitch, etc.?

**Reply by soumyaDghosh** [reply]
> In order to support video and audio output and every possible codec, you should have:
> 
> ```shell
> gstreamer1.0-plugins-base gstreamer1.0-plugins-good gstreamer1.0-plugins-bad gstreamer1.0-plugins-ugly gstreamer1.0-libav gstreamer1.0-gl gstreamer1.0-pulseaudio
> ```

I have built the gstreamer from source, with patches of yours. This is the repo,(sorry for keeping it private, for very significant reasons, it's)
https://github.com/soumyaDghosh/clapper-snap

I guess it have everything, you can check the actions, for the building to see, what's installed or what not.

> video from some webpage like YouTube, Twitch, etc.?

That's a video from youtube

**Reply by Rafostar** [reply]
You are using way too many patches. Instead of trying to do what was done here when last version was released, try checking out what is currently done (and what versions are built) on Flathub for reference: [https://github.com/flathub/com.github.rafostar.Clapper](https://github.com/flathub/com.github.rafostar.Clapper). The patches there are reduced since I build latest GStreamer 1.22 which has most problems fixed and quite a few of my fixes where merged upstream too.

Ideally you should aim for NOT building GStreamer and FFmpeg yourself, I do it cause I patch it, but with each version amount of really necessary patches goes down, so I eventually plan to stop doing that once new enough version of it is included in Flatpak.

Also, since GStreamer devs are not FFmpeg devs, its usually better/safer to not use latest possible FFmpeg as GStreamer devs always need to catch-up with latest API changes. This also goes for `gst-libav-stop-caching-codecs.patch` where both the FFmpeg install path and versions of .so files will likely be different in your build, so you should try to update it accordingly.

**Reply by soumyaDghosh** [reply]
The pathces in the master branch still seems to many, and I think the issue is with alsa, cause, inside the snap shell, when I run 
`ffmpeg -i Achyutam\ Keshavam\ _\ \ Armonian\ _\ Achyutashtakam.mp3 -f alsa default` 

I get 
```
Requested output format 'alsa' is not a suitable output format
default: Invalid argument
```

So, I guess it has something to do with the alsa libs.

**Reply by Rafostar** [reply]
I have written my honest opinion about broken Clapper Snap package made by someone else in the meantime while you were trying to finish your own build. I am giving a link to it here as a proof that it was indeed me in that [discussion]( https://forum.snapcraft.io/t/request-for-transfer-of-ownership-clapper/35320/3?u=rafostar).

**Reply by soumyaDghosh** [reply]
Thank you dev @Rafostar

**Reply by soumyaDghosh** [reply]
I tried my best to keep the conversation civil, as much as I can.
